File: sample-app-java/src/main/java/com/weedow/spring/data/search/sample/java/entity/Vehicle.java
package com.weedow.spring.data.search.sample.java.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import javax.persistence.*;

@Entity
public class Vehicle extends JpaPersistable<Long> {

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private VehicleType vehicleType;

    @Column(nullable = false)
    private String brand;

    @Column(nullable = false)
    private String model;

    @ManyToOne(optional = false)
    @JsonIgnoreProperties({"vehicles"})
    private Person person;

    public VehicleType getVehicleType() {
        return vehicleType;
    }

    public Vehicle setVehicleType(VehicleType vehicleType) {
        this.vehicleType = vehicleType;
        return this;
    }

    public String getBrand() {
        return brand;
    }

    public Vehicle setBrand(String brand) {
        this.brand = brand;
        return this;
    }

    public String getModel() {
        return model;
    }

    public Vehicle setModel(String model) {
        this.model = model;
        return this;
    }

    public Person getPerson() {
        return person;
    }

    public Vehicle setPerson(Person person) {
        this.person = person;
        return this;
    }
}

File: core/src/test/kotlin/com/weedow/spring/data/search/example/model/Vehicle.kt
package com.weedow.spring.data.search.example.model

import javax.persistence.*

@Entity
class Vehicle(
        @Column(nullable = false)
        @Enumerated(EnumType.STRING)
        val vehicleType: VehicleType,

        @Column(nullable = false)
        val brand: String,

        @Column(nullable = false)
        val model: String,

        @ManyToOne
        val person: Person

) : JpaPersistable<Long>() {


}

enum class VehicleType {
    CAR, MOTORBIKE, SCOOTER, VAN, TRUCK
}
